engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -17,9 +18,40 @@ def self.update_cache(remote_ip, system_login, product_id, registry: false)
     )
   end
 
+  def self.update_cache_not_payg(cache_key, mode)
```

Comment:
Why do we have to differentiate between payg and other flavors for the cache update? Appears to me to be a fully generic mechanism, we write a given key to a given location.